<a href="https://colab.research.google.com/github/matthewpecsok/data_engineering/blob/main/tutorials/de_tutorial_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import sqlite3

In [ ]:
!wget -O movies.db https://github.com/matthewpecsok/data_engineering/blob/main/data/movies.sqlite?raw=true

con = sqlite3.connect('movies.db')

%load_ext sql
%sql sqlite:///movies.db

%config SqlMagic.autopandas = True
%config SqlMagic.feedback = False

--2023-01-31 23:45:41--  https://github.com/matthewpecsok/data_engineering/blob/main/data/movies.sqlite?raw=true
Resolving github.com (github.com)... 20.27.177.113
Connecting to github.com (github.com)|20.27.177.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://github.com/matthewpecsok/data_engineering/raw/main/data/movies.sqlite [following]
--2023-01-31 23:45:42--  https://github.com/matthewpecsok/data_engineering/raw/main/data/movies.sqlite
Reusing existing connection to github.com:443.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/matthewpecsok/data_engineering/main/data/movies.sqlite [following]
--2023-01-31 23:45:42--  https://raw.githubusercontent.com/matthewpecsok/data_engineering/main/data/movies.sqlite
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|

In [ ]:
pd.read_sql_query('select * from movies',con)

id                            original_title     budget  popularity  \
0     43597                                    Avatar  237000000         150   
1     43598  Pirates of the Caribbean: At World's End  300000000         139   
2     43599                                   Spectre  245000000         107   
3     43600                     The Dark Knight Rises  250000000         112   
4     43601                               John Carter  260000000          43   
...     ...                                       ...        ...         ...   
4768  48395                               El Mariachi     220000          14   
4769  48396                                 Newlyweds       9000           0   
4770  48397                 Signed, Sealed, Delivered          0           1   
4771  48398                          Shanghai Calling          0           0   
4772  48399                         My Date with Drew          0           1   

     release_date     revenue                                     title  \
0      2009-12-10  2787965087                                    Avatar   
1      2007-05-19   961000000  Pirates of the Caribbean: At World's End   
2      2015-10-26   880674609                                   Spectre   
3      2012-07-16  1084939099                     The Dark Knight Rises   
4      2012-03-07   284139100                               John Carter   
...           ...         ...                                       ...   
4768   1992-09-04     2040920                               El Mariachi   
4769   2011-12-26           0                                 Newlyweds   
4770   2013-10-13           0                 Signed, Sealed, Delivered   
4771   2012-05-03           0                          Shanghai Calling   
4772   2005-08-05           0                         My Date with Drew   

      vote_average  vote_count  \
0              7.2       11800   
1              6.9        4500   
2              6.3        4466   
3              7.6        9106   
4              6.1        2124   
...            ...         ...   
4768           6.6         238   
4769           5.9           5   
4770           7.0           6   
4771           5.7           7   
4772           6.3          16   

                                               overview  \
0     In the 22nd century, a paraplegic Marine is di...   
1     Captain Barbossa, long believed to be dead, ha...   
2     A cryptic message from Bond’s past sends him o...   
3     Following the death of District Attorney Harve...   
4     John Carter is a war-weary, former military ca...   
...                                                 ...   
4768  El Mariachi just wants to play his guitar and ...   
4769  A newlywed couple's honeymoon is upended by th...   
4770  "Signed, Sealed, Delivered" introduces a dedic...   
4771  When ambitious New York attorney Sam is sent t...   
4772  Ever since the second grade when he first saw ...   

                                                tagline     uid  director_id  
0                           Enter the World of Pandora.   19995         4762  
1        At the end of the world, the adventure begins.     285         4763  
2                                 A Plan No One Escapes  206647         4764  
3                                       The Legend Ends   49026         4765  
4                  Lost in our world, found in another.   49529         4766  
...                                                 ...     ...          ...  
4768  He didn't come looking for trouble, but troubl...    9367         5097  
4769  A newlywed couple's honeymoon is upended by th...   72766         6485  
4770                                               None  231617         7108  
4771                           A New Yorker in Shanghai  126186         7109  
4772                                               None   25975         7110  

[4773 rows x 13 columns]

for this scenario say we want to move all the movies into 
1. flat file store 
2. a new database


to do this we'd like to move them in batches of 100 movies at a time. 

how many batches of data will we need to move?

there are 4773 rows of data. 100 movies at a time would mean we will move 48 batches of data, with the last batch having 73 records instead of 100.


